observability/src/main/java/io/grpc/observability/interceptors/ConfigFilterHelper.java

```diff
@@ -0,0 +1,217 @@
+   */
+  public static ConfigFilterHelper factory(ObservabilityConfig config) {
+    ConfigFilterHelper filterHelper = new ConfigFilterHelper(config);
+    if (config.isEnableCloudLogging()) {
+      filterHelper.setMethodOrServiceFilterMaps();
+      filterHelper.setEventFilterSet();
+    }
+    return filterHelper;
+  }
+
+  @VisibleForTesting
+  void setMethodOrServiceFilterMaps() {
+    List<LogFilter> logFilters = config.getLogFilters();
+    if (logFilters == null) {
+      return;
+    }
+
+    boolean globalLog = false;
+    FilterParams globalParams = null;
+    Map<String, FilterParams> perServiceFilters = new HashMap<>();
+    Map<String, FilterParams> perMethodFilters = new HashMap<>();
+
+    for (LogFilter currentFilter : logFilters) {
+      // '*' for global, 'service/*' for service glob, or 'service/method' for fully qualified
+      String methodOrServicePattern = currentFilter.pattern;
+      int currentHeaderBytes
+          = currentFilter.headerBytes != null ? currentFilter.headerBytes : 0;
+      int currentMessageBytes
+          = currentFilter.messageBytes != null ? currentFilter.messageBytes : 0;
+      if (methodOrServicePattern.equals("*")) {
+        // parse config for global, e.g. "*"
+        if (globalLog) {
+          logger.log(Level.WARNING, "Duplicate entry : {0}", methodOrServicePattern);
+          continue;
+        }
+        globalLog = true;
+        globalParams = FilterParams.create(true,
+            currentHeaderBytes, currentMessageBytes);
+      } else if (methodOrServicePattern.endsWith("/*")) {
+        // TODO(DNVindhya): check if service name is a valid string for a service name
+        // parse config for a service, e.g. "service/*"
+        String service = MethodDescriptor.extractFullServiceName(methodOrServicePattern);
+        if (perServiceFilters.containsKey(service)) {
+          logger.log(Level.WARNING, "Duplicate entry : {0)", methodOrServicePattern);
+          continue;
+        }
+        FilterParams params = FilterParams.create(true,
+            currentHeaderBytes, currentMessageBytes);
+        perServiceFilters.put(service, params);
+      } else {
+        // TODO(DNVVindhya): check if methodOrServicePattern is a valid full qualified method name
+        // parse pattern for a fully qualified method, e.g "service/method"
+        if (perMethodFilters.containsKey(methodOrServicePattern)) {
+          logger.log(Level.WARNING, "Duplicate entry : {0}", methodOrServicePattern);
+          continue;
+        }
+        FilterParams params = FilterParams.create(true,
+            currentHeaderBytes, currentMessageBytes);
+        perMethodFilters.put(methodOrServicePattern, params);
+      }
+    }
+    this.globalLog = globalLog;
+    this.globalParams = globalParams;
+    this.perServiceFilters = ImmutableMap.copyOf(perServiceFilters);
+    this.perMethodFilters = ImmutableMap.copyOf(perMethodFilters);
+    this.methodOrServiceFilterPresent = true;
```

Comment:
just realized that this is still set to `'true` even when `logFilters` is empty or when `perServiceFilters` and `perMethodFilters` are empty. Isn't that inefficient?

Reply:
Added check to set `methodOrServiceFilterPresent` to `true`, only if either of the maps are non empty.